### PR TITLE
Support for count() method for string

### PR DESCRIPTION
**Pull request by rdesai16**
Full functionality implementation for count() method.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the PR. It'd be great if you could please only include the patches relevant to `str.count()` in the changeset, thanks.

**Comment by rdesai16** [comment]
Yes I will keep that in mind. Sorry for the confusion. Somehow it had my other branch changes which I didn't see in diff preview. I will soon submit relevant changes with other irrelevant changes removed.

**Comment by rdesai16** [comment]
Hi @stuartarchibald, This PR is ready for the review now.

**Comment by esc** [comment]
@rdesai16 thanks for contributing to Numba. I had a quick look at this and it would seem that the changes (or some changes) from #4049 are still included in this Pull-Request:

https://github.com/numba/numba/pull/4126/files

**Comment by rdesai16** [comment]
> @rdesai16 thanks for contributing to Numba. I had a quick look at this and it would seem that the changes (or some changes) from #4049 are still included in this Pull-Request:
> 
> https://github.com/numba/numba/pull/4126/files

Oh I see! I apologize for not checking that before. I thought thta I had removed unnecessary changes but looks like I haven't. Thanks for bringing that into my attention. I'll fix that.

**Comment by rdesai16** [comment]
The irrelevant changes from other branch have been removed now.

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -80,6 +80,10 @@ def find_usecase(x, y):
     return x.find(y)
 
 
+def count_usecase(x, y):
+    return x.find(y)
```
Comment (ehsantn) [inline numba/tests/test_unicode.py]: Copy-paste error: `find` -> `count`.

**Comment by stuartarchibald** [comment]
@rdesai16 Seems like this still contains code from other patches/branches. Perhaps it would be best to just close this PR, create a new branch from master and just copy in the few bits of code relevant for implementing this feature. Either that or @esc or I can go and do surgery on your branch if that would help?

**Comment by rdesai16** [comment]
@stuartarchibald I think closing this PR and submitting a new one from another branch makes sense. If you approve of that approach then I'll go ahead and do that.

**Comment by stuartarchibald** [comment]
@rdesai16 I think that might be most efficient given there's been a couple of attempts at fixing this already, feel free to do as you see fit.

**Pull request closed without merging**